gobblin-utility/src/main/java/gobblin/util/io/BatchedMeterDecorator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.util.io;
+
+import com.codahale.metrics.Meter;
+
+import gobblin.util.Decorator;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+
+/**
+ * A decorator to a {@link Meter} that batches updates for performance.
+ */
+@NotThreadSafe
+public class BatchedMeterDecorator implements Decorator {
```

Comment:
Two reasons: it is slightly harder to make it thread-safe (requires some locks), and this is intended as a high performance class, locks / synchronized blocks reduce performance.